File: php/spaceport/src/SpaceportDepartureBoard.php
<?php

declare(strict_types=1);

namespace Spaceport;

use Exception;

class SpaceportDepartureBoard
{
    private $launch_list = [];

    /**
     * SpaceportDepartureBoard constructor.
     * @param ISpacelineLaunchInfoProvider
     */
    public function __construct(ISpacelineLaunchInfoProvider $provider)
    {
        $this->launch_list = $provider->getCurrentLaunches();
        $this->sortLaunches();
    }

    /**
     * @return array
     */
    public function getLaunchList(): array
    {
        return $this->launch_list;
    }

    private function sortLaunches()
    {
        throw new Exception('TODO - implement the sorting logic for the Spaceport Departure Board');
    }
}

File: python/spaceport/spaceport_departure_board.py
class SpaceportDepartureBoard():
    def __init__(self, launch_info_provider):
        self.launch_list = launch_info_provider.get_current_launches()
        self.sort_launches()

    def get_launch_list(self):
        return self.launch_list

    def sort_launches(self):
        # TODO - implement the sorting logic!
        pass